### PR TITLE
Display style survey

### LINKED ISSUE #9: Style Survey
Before adding a monthly subscription I would like to choose my style preferences to ensure that I get clothes that better match my personal style. When I visit /crates, the first time I click on + subscribe I am taken to a style preferences page /style-preferences. I am presented with a style survey that consists of a series of categories such as tops, bottoms, dresses, shoes, accessories, etc. For each category I am given several images to choose from with instructions to choose the images that best reflects my style. After I have chosen the images from each category I am given a summary of my style such as ‘classic but edgy’.

### LINKED ISSUE #17: Style Survey Algorithm
Need to figure out a way to assign a style, based off a respondents answers.

I found a decision tree, which I think is in the right ballpark for how to accomplish this. My recommendation would be to start out with a smaller survey to just get it working.

### LINKED ISSUE #19: Create StyleSurvey component
A <StyleSurvey/>  component needs to be created, that will display the form

### LINKED ISSUE #20: - Update component  state with class methods
- When the user selects an image from a question,  the click event should register the event and get the data associated with it.

### LINKED ISSUE #21: -Add functionality  to + subscription button
When the user click on the `+Subscribe` button the user should be redirected to the following route: `/style-preferences`

### PR DESCRIPTION
**Pull request by Atos20**
PR Template:
What’s this PR do?  

-  it displays the `<StyleSurvey/>` component
- it adds a new path to the user's paths
- it redirects the user to `/style-preferences` path when the user clicks on the`+Subscribe` button
- it styles  the StyleSurvey component
- it reuses UI component to create the Style survey
- It adds a new case to the user reducer
- it adds a new property to the store state to keep track of the user has completed the survey
- it adds a pull request to the project repository

Where should the reviewer start?  

-  cd into the `Crate/code/web/src` and open 
    - /modules/StyleSurvey/StyleSurvey.js
    - /modules/crate/Item.js 
    - /modules/crate/List.js
    - /modules/user/api/actions.js
    - /setup/store.js

How should this be manually tested?  
- cd into `code/web` and run `npm start` to start the server on `localhost:3000` 
- cd into `code/api` and run `npm start` to start the server on `localhost:8000` \
- make sure the files have finished compiling
- log in as a user
- navigate to `/crates`
- select a subscription

Any background context you want to provide?  
 - As a team, we have been assigned with creating a style survey that a user can fill out and find out the unique style.
 - Our team is conformed of 4BE students that will be modifying the existing schemas built in the project
 
What are the relevant tickets?  
 - closes  #19 
 - closes #9 
 - closes #17 
 - closes #20 
 - closes #21 

Screenshots (if appropriate)  

 Screen Shot 2020-12-07 at 10 37 38 PM

### CODE REVIEW

File: code/web/src/modules/user/api/state.js [hunk 1]
```diff
@@ -7,12 +7,20 @@ export const userInitialState = {
...
   isAuthenticated: false,
-  details: null
+  details: null,
+  isSurveyCompleted: false
 }
 
 // State
 export default (state = userInitialState, action) => {
   switch (action.type) {
+    case ASIGN_STYLE:
```
Comment (npdarrington) [inline code/web/src/modules/user/api/state.js]: We can change the naming of this to 'ASSIGN'

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+PR Template:
```
Comment (npdarrington) [inline pull_request_template.md:1]: Awesome including a PR Template. Thank you!

File: code/web/src/ui/image/Tile.js [hunk 3]
```diff
@@ -4,15 +4,17 @@ import PropTypes from 'prop-types'
...
   return (
-    <div style={Object.assign({ height, width }, style)} {...others}>
+    <div style={Object.assign({ height, width , margin}, style)} {...others}>
       {children}
 
       {/* language=CSS */}
       <style jsx>{`
         div {
+          background-position: center; 
+          background-repeat: no-repeat;
```
Comment (npdarrington) [inline code/web/src/ui/image/Tile.js:17]: While these 2 properties are working, they seem to be causing some extra blank space above and below the picture.

**Review by npdarrington — COMMENTED** [review]
Overall looks really good and we seem to be in a good spot to do some minor adjustments and begin testing while adding new functionality. Two things I noticed listed below.

1. Rebase was not used, are we supposed to be using rebase for everything with this project or doing it as normal?
2. There is code that replaces some of the existing code base to use redux dev tools, however there was no package.json or package-lock.json that a user pulling this branch could pull down and run `npm i` to get all of the new dependencies installed. Let's add those files in so we can install the dependencies for them.

File: code/web/src/modules/user/api/state.js (hunk 1 above)
Comment (Atos20) [reply code/web/src/modules/user/api/state.js]: change it from `ASIGN` to just `ASSIGN` or `ASIGN` to `ASSIGN_STYLE`?

File: code/web/src/ui/image/Tile.js (hunk 3 above)
Comment (Atos20) [reply code/web/src/ui/image/Tile.js:17]: I will take a closer look at this today and figure out why.

**Pull request closed without merging**